pulsar-broker/src/test/java/org/apache/pulsar/s3offload/OffloadIndexTest.java

```diff
@@ -0,0 +1,196 @@
+
+        assertTrue(entry2.getEntryId() == 100L);
+        assertTrue(entry2.getPartId() == 3);
+        assertTrue(entry2.getOffset() == 1234L);
+    }
+
+
+    // use mock to setLastEntryId
+    class LedgerMetadataMock extends org.apache.bookkeeper.client.LedgerMetadata {
+        long lastId = 0;
+        public LedgerMetadataMock(int ensembleSize, int writeQuorumSize, int ackQuorumSize, org.apache.bookkeeper.client.BookKeeper.DigestType digestType, byte[] password, Map<String, byte[]> customMetadata, boolean storeSystemtimeAsLedgerCreationTime) {
+            super(ensembleSize, writeQuorumSize, ackQuorumSize, digestType, password, customMetadata, storeSystemtimeAsLedgerCreationTime);
+        }
+
+        @Override
+        public long getLastEntryId(){
+            return  lastId;
+        }
+
+        public void setLastEntryId(long lastId) {
+            this.lastId = lastId;
+        }
+    }
+
+    private LedgerMetadata createLedgerMetadata() throws Exception {
+
+        Map<String, byte[]> metadataCustom = Maps.newHashMap();
+        metadataCustom.put("key1", "value1".getBytes(UTF_8));
+        metadataCustom.put("key7", "value7".getBytes(UTF_8));
+
+        ArrayList<BookieSocketAddress> bookies = Lists.newArrayList();
+        BookieSocketAddress BOOKIE1 = new BookieSocketAddress("127.0.0.1:3181");
+        BookieSocketAddress BOOKIE2 = new BookieSocketAddress("127.0.0.2:3181");
+        BookieSocketAddress BOOKIE3 = new BookieSocketAddress("127.0.0.3:3181");
+        bookies.add(0, BOOKIE1);
+        bookies.add(1, BOOKIE2);
+        bookies.add(2, BOOKIE3);
+
+        LedgerMetadataMock metadata = new LedgerMetadataMock(3, 3, 2,
+            DigestType.CRC32C, "password".getBytes(UTF_8), metadataCustom, false);
+
+        metadata.addEnsemble(0, bookies);
+        metadata.setLastEntryId(5000);
+        return metadata;
+    }
+
+    // prepare metadata, then use builder to build a OffloadIndexBlockImpl
+    // verify get methods, readout and fromStream methods.
+    @Test
+    public void offloadIndexBlockImplTest() throws Exception {
+        OffloadIndexBlockBuilder blockBuilder = OffloadIndexBlockBuilder.create();
+        LedgerMetadata metadata = createLedgerMetadata();
+        log.debug("created metadata: {}", metadata.toString());
+
+        blockBuilder.withMetadata(metadata);
+
+        blockBuilder.addBlock(0, 2, 0);
+        blockBuilder.addBlock(1000, 3, 64 * 1024 * 1024);
+        blockBuilder.addBlock(2000, 4, 64 * 1024 * 1024);
+        OffloadIndexBlock indexBlock = blockBuilder.build();
+
+        // verify getEntryCount and getLedgerMetadata
+        assertTrue(indexBlock.getEntryCount() == 3);
+        assertTrue(indexBlock.getLedgerMetadata() == metadata);
+
+        // verify getIndexEntryForEntry
+        OffloadIndexEntry entry1 = indexBlock.getIndexEntryForEntry(0);
+        assertTrue(entry1.getEntryId() == 0 && entry1.getPartId() == 2 && entry1.getOffset() == 0);
```

Comment:
use assertEquals for each of these. If the test fails, we want to be able to see why without having to spin up a debugger, or rewrite the test.